### PR TITLE
Minimum and maximum time updates

### PR DESCRIPTION
**Pull request by bbengfort**
Updated earliest to use a `MINIMUM_TIME` constant and latest to use the
`MAXIMUM_TIME` constant. Created a new method `stream.current` that looks
up the point closest to now (or earlier).

### CODE REVIEW

File: btrdb/stream.py [hunk 1]
```diff
@@ -771,6 +794,31 @@ def latest(self):
...
+        None
+
+        Returns
+        -------
+        tuple
+            The latest points of data found among all streams
+        """
+        latest = []
+        params = self._params_from_filters()
         start = params.get("end", currently_as_ns())
```
Comment (bbengfort) [inline btrdb/stream.py]: Maybe this should ignore the filters if we're looking for the point closest to now?

File: btrdb/stream.py [hunk 2]
```diff
@@ -780,6 +802,37 @@ def latest(self):
...
+        params = self._params_from_filters()
+        now = currently_as_ns()
+        end = params.get("end", None)
+        start = params.get("start", None)
+
+        if (end is not None and end <= now) or (start is not None and start > now):
+            raise ValueError("current time is not included in filtered stream range")
+
+        for s in self._streams:
+            version = self.versions()[s.uuid]
```
Comment (bbengfort) [inline btrdb/stream.py]: I'm starting to think we should make version a filter constant that defaults to 0 - if we use the latest version every time we will always be excluding points that are in the PQM, which means queries will not get the latest, buffered data.

Comment (looselycoupled) [reply btrdb/stream.py]: I didnt realize that... I was under the impression that 0 will still check the buffer.  Good to know.

File: btrdb/stream.py [hunk 3]
```diff
@@ -196,12 +198,14 @@ def collection(self):
...
-        Returns the first point of data in the stream.  Returns None if error
+        Returns the first point of data in the stream. Returns None if error
         encountered during lookup or no values in stream.
 
         Parameters
         ----------
-        None
+        version : int, default: 0
+            Specify the version of the stream to query; if zero, queries the latest
+            stream state rather than pinning to a version.
```
Comment (looselycoupled) [inline btrdb/stream.py]: Surprised I missed this

File: btrdb/stream.py (hunk 2 above)
Comment (looselycoupled) [reply btrdb/stream.py]: It does kind of make sense to keep this in the filter code...  Let's sit on the idea for a little bit

File: btrdb/version.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 __version_info__ = {
     'major': 5,
     'minor': 5,
-    'micro': 0,
+    'micro': 1,
```
Comment (looselycoupled) [inline btrdb/version.py]: I normally reserve this for a release branch in this repo.  What was your reasoning for incrementing version in this PR?

**Comment by looselycoupled** [comment]
This whole PR reminds me that `latest`, `earliest`, and now `current` should have an explicit section ini the docs somewhere (besides the stream API page).

File: btrdb/stream.py (hunk 2 above)
Comment (bbengfort) [reply btrdb/stream.py]: I'm 95% sure that we don't get a version until the flush to the object store, new versions only happen a handful of times an hour.

File: btrdb/version.py (hunk 4 above)
Comment (bbengfort) [reply btrdb/version.py]: I needed to debug which version I was importing, I've reset it back to 0.

**Comment by bbengfort** [comment]
@looselycoupled I've modified the version number and created issues based on the other comments; let me know when you're ready to merge this in.

**Pull request merged**